File: week13/CodeKata13.py
class Mouth:
    def speech(self):
        return "HELLO HELLO"

class Lung:
    def inhaleAir(self):
        return "Air inhaled"
    def exhaleAir(self):
        return "Air exhaled"

class Feet:
    def step(self):
        return "Step"

class Person:
    def __init__(self):
        self.myMouth = Mouth()
        self.myLung = Lung()
        self.myFeet = Feet()

    def talk (self):
        return self.myMouth.speech()
    def breath (self):
        message = self.myLung.inhaleAir();
        return message + " " + self.myLung.exhaleAir()
    def walk (self):
        message = "";
        for i in range (0,10):
            message = message + " " + self.myFeet.step()
        return message

class TestPerson:
    def __init__(self):
        self.person = Person()

    def test (self):
        if (self.person.talk() != "HELLO HELLO"):
            print "Test failed!"
        if (self.person.breath() != "Air inhaled Air exhaled"):
            print "Test failed!"
        if (self.person.walk() != " Step Step Step Step Step Step Step Step Step Step"):
            print "Test failed!"

test = TestPerson()
test.test()

File: week13/CodeKata13.js
var Mouth = function () {
};
Mouth.prototype.speech = function() {
    return "HELLO HELLO";
};

var Lung = function () {
};
Lung.prototype.inhaleAir = function () {
    return "Air inhaled";
};
Lung.prototype.exhaleAir = function () {
    return "Air exhaled";
};

var Feet = function () {

};
Feet.prototype.step = function () {
    return "Step";
};

var Person = function () {
    this.myMouth = new Mouth();
    this.myLung = new Lung();
    this.myFeet = new Feet();
};

Person.prototype.talk = function() {
    return this.myMouth.speech();
};
Person.prototype.breath = function() {
    var message = this.myLung.inhaleAir();
    return message + " " + this.myLung.exhaleAir();
};
Person.prototype.walk = function() {
    var message = "";
    for (var i=0; i<10; i++) {
        message = message + " " + this.myFeet.step();
    }
    return message;
};

var TestPerson = function () {
    this.person = new Person();
};
TestPerson.prototype.test = function() {
    if (this.person.talk() !== "HELLO HELLO") {
        console.log("Test failed!");
    }
    if (this.person.breath() !== "Air inhaled Air exhaled") {
        console.log("Test failed!");
    }
    if (this.person.walk() !== " Step Step Step Step Step Step Step Step Step Step") {
        console.log("Test failed!");
    }
};

test = new TestPerson();
test.test();